libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c

```diff
@@ -1112,321 +1109,343 @@ uint16_t usType = 0U;
+								{
+									( void ) prvProcessDNSCache( pcName, &ulIPAddress, pxDNSAnswerRecord->ulTTL, pdFALSE );
+                  usNumARecordsStored++;    /* Track # of A records stored */
+								}
+
+								FreeRTOS_inet_ntop( FREERTOS_AF_INET, ( const void * ) &( ulIPAddress ), cBuffer, sizeof( cBuffer ) );
+								/* Show what has happened. */
+								FreeRTOS_printf( ( "DNS[0x%04lX]: The answer to '%s' (%s) will%s be stored\n",
+												   ( UBaseType_t ) pxDNSMessageHeader->usIdentifier,
+												   pcName,
+												   cBuffer,
+												   ( xDoStore != 0 ) ? "" : " NOT" ) );
+							 }
+							 #endif /* ipconfigUSE_DNS_CACHE */
 						}
-						#endif /* ipconfigUSE_DNS_CACHE */
-					}
 
-					pucByte = &( pucByte[ sizeof( DNSAnswerRecord_t ) + sizeof( uint32_t ) ] );
-					uxSourceBytesRemaining -= ( sizeof( DNSAnswerRecord_t ) + sizeof( uint32_t ) );
-				}
-				else if( uxSourceBytesRemaining >= sizeof( DNSAnswerRecord_t ) )
-				{
-					/* It's not an A record, so skip it. Get the header location
-					and then jump over the header. */
-
-					/* MISRA c 2012 rule 11.3 relaxed as pucByte is being used in
-					various places to point to various parts of the DNS records */
-					pxDNSAnswerRecord = ipPOINTER_CAST( DNSAnswerRecord_t *, pucByte );
+						pucByte = &( pucByte[ sizeof( DNSAnswerRecord_t ) + sizeof( uint32_t ) ] );
+						uxSourceBytesRemaining -= ( sizeof( DNSAnswerRecord_t ) + sizeof( uint32_t ) );
+					}
+					else if( uxSourceBytesRemaining >= sizeof( DNSAnswerRecord_t ) )
+					{
+						/* It's not an A record, so skip it. Get the header location
+						and then jump over the header. */
+						/* Cast the response to DNSAnswerRecord for easy access to fields of the DNS response. */
+						pxDNSAnswerRecord = ipPOINTER_CAST( DNSAnswerRecord_t *, pucByte );
 
-					pucByte = &( pucByte[ sizeof( DNSAnswerRecord_t ) ] );
-					uxSourceBytesRemaining -= sizeof( DNSAnswerRecord_t );
+						pucByte = &( pucByte[ sizeof( DNSAnswerRecord_t ) ] );
+						uxSourceBytesRemaining -= sizeof( DNSAnswerRecord_t );
 
-					/* Determine the length of the answer data from the header. */
-					usDataLength = FreeRTOS_ntohs( pxDNSAnswerRecord->usDataLength );
+						/* Determine the length of the answer data from the header. */
+						usDataLength = FreeRTOS_ntohs( pxDNSAnswerRecord->usDataLength );
 
-					/* Jump over the answer. */
-					if( uxSourceBytesRemaining >= usDataLength )
-					{
-						pucByte = &( pucByte[ usDataLength ] );
-						uxSourceBytesRemaining -= usDataLength;
+						/* Jump over the answer. */
+						if( uxSourceBytesRemaining >= usDataLength )
+						{
+							pucByte = &( pucByte[ usDataLength ] );
+							uxSourceBytesRemaining -= usDataLength;
+						}
+						else
+						{
+							/* Malformed response. */
+							xReturn = pdFALSE;
+							break;
+						}
 					}
 					else
 					{
-						/* Malformed response. */
-						return dnsPARSE_ERROR;
+						/* Do nothing */
```

Comment:
Or maybe:
~~~c
    /* The IP-address found will be returned. */
~~~

Reply:
I kept it as is because the returning of IP-address depends on the value of `xExpected`. I added the explanation there.